NDTensors/src/lib/TypeParameterAccessors/src/specify_parameters.jl

```diff
@@ -0,0 +1,48 @@
+using Compat: Compat
+
+Compat.@assume_effects :foldable function specify_parameter(type::Type, pos::Int, param)
+  return _specify_parameter(parameter(type, pos), type, Position(pos), param)
+end
+
+Compat.@assume_effects :foldable function specify_parameter(
+  type::Type, pos::Position, param
+)
+  return _specify_parameter(parameter(type, pos), type, pos, param)
+end
+
+function _specify_parameter(::TypeVar, type::Type, pos::Position, param)
+  return set_parameter(type, pos, TypeParameter(param))
+end
+
+function _specify_parameter(::TypeVar, type::Type, pos::Position, param::Type)
+  return set_parameter(type, pos, param)
+end
```

Comment:
I don't see why both of these are needed, isn't this logic handled inside of `set_parameter`?

Reply:
This can be handled by set_parameter and I modified the code to work for this.